add api changes for 2.x (#588)
SAP/cloud-sdk: docs/js/features/odata/common/entity/entity-builder.mdx

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<Tabs groupId="version" defaultValue="major" values={[ { label: 'SDK 2.x', value: 'major' }, { label: 'SDK 1.x', value: 'deprecated' } ]}>
<TabItem value="major">

> In version 2.0 of the SDK, two new changes have been introduced:

Reviewer: ```suggestion
In version 2.0 of the SAP Cloud SDK, two new changes have been introduced:
```

- an API class for each entity (e.g. `BusinessPartnerApi`), providing access to its properties via a `schema`, and  
- a common utility method, e.g. `businessPartnerService()`, with accessors for all the API classes.  

> To build an entity by assigning its properties, use the entity builders provided by the api class.

Reviewer: ```suggestion
To build an entity by assigning its properties, use the entity builders provided by the API class.
```

> Use the `.entityBuilder()` method to access the builder, set the properties and finally use the `.build` method to yield the entity.

Reviewer: ```suggestion
First invoke the `.entityBuilder()` method to access the builder, then you can set the properties and finally call the `.build` method to yield the entity.
```

Reviewer: This article uses `.method`, `method()` and `.method()`. I think it would be better to make this consistent.

> To set navigation properties, that link to other entities, you have to create the linked entities using their respective builders.

Reviewer: I'm not totally sure, but this seems to read better without the comma.

```suggestion
To set navigation properties that link to other entities, you have to create the linked entities using their respective builders.
```

Reviewer: A code example would be helpful in my opinion.

> Navigation properties, that are linked through a one-to-many relation have to be assigned in an array - one-to-one relations are assigned as single objects.

Reviewer: I am not sure what you mean with "assigned in an array". I think it would be clearer if you use this:
```suggestion
Navigation properties that are linked through a one-to-many relation, are set by passing an array (even if that array only contains one entity). 
One-to-one relations are assigned as objects.
```

> The example below shows how you would create the data from above using the entity builder.

Reviewer: `data from above`? I am not sure what this refers to.

```suggestion
The example below shows how you can create the data from above using the entity builder.
```

Author: It refers to the example (for OData JSON representation) shown just above this section (its in a separate file so not very obvious)

You can also add fields, that are unknown according to the specification, if you add them as custom fields.
To achieve this, pass an object the `.withCustomFields` method, where the keys denote the names of the custom fields, and the values their respective values.

<Tabs groupId="version" defaultValue="major" values={[ { label: 'SDK 2.x', value: 'major' }, { label: 'SDK 1.x', value: 'deprecated' } ]}>
<TabItem value="major">

```ts
import {
  BusinessPartnerApi,

>   BusinessPartnerAddressApi

Reviewer 2: Where is the address used in both cases?
